### PR TITLE
My changes in Making a Authors Page

### PR DESCRIPTION
**Pull request by Aditya-eddy**
Hello Everyone,
I have created this draft PR.I have made a  author's Page which was due pending. We currently don't have an author's page. But I have created a author's page in this PR. But it probably needs some styling. I have not Created a individual author's page which Displays the author details.

### CODE REVIEW

**Comment by nehagup** [comment]
@Aditya-eddy thanks for the initiative, please lmk whenever you're ready for review.

**Comment by Aditya-eddy** [comment]
@nehagup I have marked the PR ready for review

**Comment by Aditya-eddy** [comment]
Hey @nehagup  to be updating you on the authors page. I have finally created the authors page which. which shows all the authors which have contributed to the posts. Now I have a query in making the indivisual author's page which contains the authors and the posts by the authors, should i make a seprate pr for it or just use the same. Also I am attaching the styling of the page if you have problems or changes in the styling of the website please do tell me.
ps:- sorry for writing such a big note.
[Screencast from 29-01-24 08:50:49 PM IST.webm](https://github.com/keploy/blog-website/assets/70089590/89b0175b-4f8e-4151-a6a9-79c673db9c8e)

**Review by nehagup — CHANGES_REQUESTED** [review]
Thanks for taking the initiative @Aditya-eddy. 

- We need to maintain the styling of the page same as the mentioned in the tailwind config file.  
- For the design of the page you can take reference from https://keploy-blog.hashnode.dev/members 

You can keep the same branch and PR until the first version of this page is launched and then we can make smaller PRs.

Please feel free to ask questions.

**Comment by Aditya-eddy** [comment]
Hey @nehagup I have made styling according to the website given by you. I'll attach a video with it. Please tell your approval on it.
[Uploading Screencast from 01-02-24 04:22:55 PM IST.webm…]()

**Comment by nehagup** [comment]
@Aditya-eddy please lmk when you're ready for review

**Comment by Aditya-eddy** [comment]
Hey @nehagup  i have updated the Authors page. According to the website's styling please give me your input on this.
[Screencast from 02-02-24 12:12:54 AM IST.webm](https://github.com/keploy/blog-website/assets/70089590/68a09ca7-90a4-4f66-b1e0-fe50b45f3e6d)

**Comment by Akash-Singh04** [comment]
@Aditya-eddy Let me know if you need any help with this on slack!

**Comment by Aditya-eddy** [comment]
@Akash-Singh04  I have review pending can you please review this pr.

**Comment by Akash-Singh04** [comment]
The design looks good and clean to me, maybe we can think of having two rows of authors per page instead of one, would generally be more space efficient and better UI in my opinion

image



I'm sure the relevant maintainers will be able to help you with the full review.

**Review by nehagup — CHANGES_REQUESTED** [review]
1. The page needs to light mode (like the website) in default mode - right now it's using dark bg by default. 

2. The container of the "authors listing" need to be responsive and fit to 70% of the vertical height at least, so that we can show more number of members. 

3.  The back and Next button should be disabled on first and last page respectively. 
4.  After clicking on the author name on any of the blog it should redirect to the author page with selected author. For example "If someone clicks on Animesh, It should redirect to /blog/authors/Animesh%20Pathak"

**Comment by Aditya-eddy** [comment]
@nehagup about the 4th point today this morning I opened this [#1502](https://github.com/keploy/keploy/issues/1502)
Which talks about the same issue but @Sonichigo  told me to keep this as a good-first-issue. So what will be the final verdict?

**Comment by nehagup** [comment]
> @nehagup about the 4th point today this morning I opened this [#1502](https://github.com/keploy/keploy/issues/1502) Which talks about the same issue but @Sonichigo told me to keep this as a good-first-issue. So what will be the final verdict? About the rest points I'll get back to you. Thanks for feedback!!

My apologies for the confusion here. I think we can include that issue so that the authors page that you develop can actually be used. Right now there's no direction to authors/[name]. So please go ahead, include that.

**Comment by Aditya-eddy** [comment]
@nehagup Thanks will get to it now!

**Comment by Aditya-eddy** [comment]
@nehagup I have adjusted the changes please review them. Thanks!!

**Review by nehagup — COMMENTED** [review]
1. There are build errors for authors/[slug] - please resolve them.
2. the /authors page is redirecting to home page because it's not able to resolve. 

Rest looks good!

**Review by nehagup — COMMENTED** [review]
- Author title needs to be smaller text. Align /blog/community  "More stories"
- Author title and the column of author lists needs to be in the same ident.
- the disabled button "Back/Next" is generally lighest color to avoid attention. 
- The pointer should be changed while hovering over the authors individually to show it's a link.

**Comment by Aditya-eddy** [comment]
Hey @nehagup I'll attach the screenshots of the text and identation change. I have also change the Author title according to the More stories. Please give you feedback on it. Thanks!!
Screenshot from 2024-02-09 09-08-57
Screenshot from 2024-02-09 09-08-43
Screenshot from 2024-02-09 09-08-32

**Review by nehagup — COMMENTED** [review]
just the pointer should be changed while hovering over the authors individually to show it's a link.

Rest looks good! 👏

**Comment by nehagup** [comment]
@Aditya-eddy just checked that the build is failing for 2 pages.

**Comment by Aditya-eddy** [comment]
@nehagup what were the pages?

**Review by nehagup — COMMENTED** [review]
npm build is failing with 
```
 Linting and checking validity of types  .Failed to compile.

./pages/authors/[slug].tsx:14:8
Type error: Type '{ children: Element[]; preview: any; }' is missing the following properties from type '{ preview: any; children: any; featuredImage: any; Title: any; Description: any; }': featuredImage, Title, Description

  12 |   return (
  13 |     <div className="bg-accent-1">
> 14 |       <Layout preview={preview}>
     |        ^
  15 |         <Header />
  16 |         <Container>
  17 |           <h1 className="bg-gradient-to-r from-orange-200 to-orange-100 bg-[length:100%_20px] bg-no-repeat bg-left-bottom w-max mb-8 text-4xl heading1 md:text-6xl sm:xl font-bold tracking-tighter leading-tight">

```
and 2 pages that broke during build with your last merged PR are 

 ```
> Export encountered errors on following paths:
	/community/[slug]
	/technology/[slug]
```

**Comment by Aditya-eddy** [comment]
Hey @nehagup is probably because of this pr that we merged today will correct and submit by the EOD.

**Comment by Aditya-eddy** [comment]
Hey @nehagup hopefully I have solved all the issue including the pointer. If I have not solved the cursor pointer please tell me which browser is used by you. I am using Chrome. I have also added to scale-to-105 on the authors column. I'll also attach the screenshots.
Will be waiting on the feedback.
Thanks!!
Screenshot from 2024-02-09 15-42-56

**Review by nehagup — COMMENTED** [review]
The pointer appears fine now. 

Just need to resolve the build error


```
> Export encountered errors on following paths:
	/authors/[slug]
	/community/[slug]
	/technology/[slug]
```

File: pages/technology/[slug].tsx
Comment (nehagup) [inline pages/technology/[slug].tsx:49]: Great! Build passed! Last change! Please change this to  {`${post?.title} | <blog Title>`}

**Review by nehagup — CHANGES_REQUESTED** [review]

**Review by nehagup — APPROVED** [review]

**Pull request merged**
